website/src/panel-layout.tsx

```diff
@@ -0,0 +1,267 @@
+      />
+
+      {props.word.englishGloss[0] !== "" && (
+        <props.panelType word={props.word} features={["englishGloss"]} />
+      )}
+    </>
+  )
+
+  // Contains a component rendering a word's commentary.
+  const commentaryContent = (
+    <props.panelType word={props.word} features={["commentary"]} />
+  )
+
+  return (
+    <>
+      {/* Renders audio recording. */}
+      {/* TODO: CAUSES ERRORS */}
+      <AudioPanel segment={props.word} />
+
+      <CollapsiblePanel
+        title={"Phonetics"}
+        content={phoneticsContent}
+        icon={
+          <MdRecordVoiceOver
+            size={24}
+            className={css.wordPanelButton.colpleft}
+          />
+        }
+      />
+
+      {/* If there are no segments, does not display Word Parts panel */}
+      {props.word.segments.length > 0 && (
+        <CollapsiblePanel
+          title={"Word Parts"}
+          content={wordPartsContent}
+          icon={
+            <IoEllipsisHorizontalCircle
+              size={24}
+              className={css.wordPanelButton.colpleft}
+            />
+          }
+        />
+      )}
+
+      {/* If there is no commentary, does not display Commentary panel */}
+      {props.word.commentary && (
+        <CollapsiblePanel
+          title={"Commentary"}
+          content={commentaryContent}
+          icon={<MdNotes size={24} className={css.wordPanelButton.colpleft} />}
+        />
+      )}
+    </>
+  )
+}
+
+export const VerticalMorphemicSegmentation = (p: {
+  panelType: typeof EditWordPanel | typeof WordPanel
+  segments: FormFieldsFragment["segments"]
+}) => {
+  if (!p.segments) {
+    return null
+  }
+
+  let segmentCount = p.segments.length
+
+  return (
+    <table className={css.tableContainer}>
+      <tbody>
+        {p.panelType === WordPanel
+          ? p.segments.map((segment, index) => (
+              <tr key={index}>
+                <td className={css.tableCells}>
+                  {index > 0 ? "-" : null}
+                  {segment.morpheme}
+                  {index !== segmentCount - 1 ? segment.nextSeparator : null}
+                </td>
+                <td className={css.tableCells}>
+                  {segment.matchingTag
+                    ? segment.matchingTag.title
+                    : segment.gloss}
+                </td>
+              </tr>
+            ))
+          : p.segments.map((segment, index) => (
+              <tr key={index} style={{ display: "flex" }}>
+                <td className={css.editMorphemeCells}>
+                  <p.panelType
+                    features={["segments", index.toString(), "morpheme"]}
+                  />
+                </td>
+                <td className={css.editMatchingCells}>
+                  {segment.matchingTag?.__typename === "TagForm" ? (
+                    <p.panelType
+                      features={[
+                        "segments",
+                        index.toString(),
+                        "matchingTag",
```

Comment:
Also, I think you're misunderstanding what `matchingTag` is. That's not inherent data, that's calculated as the glossary entry which provides a definition for this morpheme. Take a look at the `matching_tag` resolver in Rust, and the doc comment for it, passing this field will do nothing. I mentioned editing the morphemic segmentation as being another in-depth project because assigning morphemes a `matchingTag` means some kind of auto-complete drop-down that shows all available matching tags. That's not in scope for this PR I'd say but I would at least avoid trying to change the matchingTag fields.

Reply:
Got it, for now I'll leave that part for another time.